### PR TITLE
[nanogui] Upgrade from 2019-09-23 to 2021-11-24

### LINKED ISSUE #17052: [nanogui] update to 16/10/2020 - also it's broken
Library name: nanogui

New version number:

Other information that may be useful (release notes, etc...)

https://github.com/mitsuba-renderer/nanogui

---

The current version doesn't seem to install the headers anywhere, so it doesn't actually build.

### `CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.0)
cmake_policy(SET CMP0048 NEW)
project(pp VERSION 0.0.0 LANGUAGES CXX)

set(src src/main.cpp)

find_package(nanovg CONFIG REQUIRED)

add_executable(${PROJECT_NAME} ${src})
target_link_libraries(${PROJECT_NAME} PRIVATE nanovg::nanovg)
````

### `src/main.cpp`
https://raw.githubusercontent.com/mitsuba-renderer/nanogui/5555031/src/example2.cpp

### `shell`
```sh
$ [path]/vcpkg install nanogui
$ mkdir build && cd $_
$ cmake .. -DCMAKE_TOOLCHAIN_FILE=[path]/vcpkg/scripts/buildsystems/vcpkg.cmake
$ cmake --build .
fatal error: 'nanogui/nanogui.h' file not found
#include <nanogui/nanogui.h>
         ^~~~~~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by SamuelMarks**
**Describe the pull request**

- What does your PR fix? Fixes #17052 

- Which triplets are supported/not supported? Have you updated the CI baseline?

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### `CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.0)
cmake_policy(SET CMP0048 NEW)
project(pp VERSION 0.0.0 LANGUAGES CXX)

set(CMAKE_CXX_STANDARD 17)

file(DOWNLOAD https://raw.githubusercontent.com/mitsuba-renderer/nanogui/0146a88/src/example2.cpp src/main.cpp
    EXPECTED_HASH SHA256=3e0a2c44fd1032862ee4e969f9bdaea77ffa1cabc31a31231cf5bc8d85f0d4bc)

set(src ${CMAKE_CURRENT_BINARY_DIR}/src/main.cpp)

find_package(nanovg CONFIG REQUIRED)

add_executable(${PROJECT_NAME} ${src})
target_link_libraries(${PROJECT_NAME} PRIVATE nanovg::nanovg)
```

This doesn't work yet, but I've given it a go. Here is a single file `CMakeLists.txt` to replicate^

PS: 
```
SHA512 should be 0b0d0d077079e1a7186ec2cc640dbd48aebcefa0e1e0caa476128c6396b33d236cb5e8aaada4e1c8868a5d4aaefd90178206dd1569abab63c84af26d3309e2a5
```

### CODE REVIEW

File: ports/nanogui/CONTROL [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Source: nanogui
-Version: 2019-09-23
-Port-Version: 1
-Homepage: https://github.com/wjakob/nanogui
+Version: 2020-10-16
+Port-Version: 2
```
Comment (JonLiu1993) [inline ports/nanogui/CONTROL:3]: @SamuelMarks ,we should delete "Port-Version" or set "Port-Version" to 0

**Comment by JonLiu1993** [comment]
After completing the modification, please run _vcpkg-add-version nanogui --overwrite-version_command and submit again

**Comment by JonLiu1993** [comment]
@SamuelMarks ,please take a look:
````
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:105 (message):
  

  File does not have expected hash:

          File path: [ D:/downloads/temp/mitsuba-renderer-nanogui-0146a88b2214cd5c5c29e6dfa8d3d3d0e9ab6d9d.tar.gz ]
      Expected hash: [ 36c93bf977862ced2df4030211e2b83625e60a11fc9fdb6c1f2996bb234758331d3f41a7fbafd25a5bca0239ed9bac9c93446a4a7fac4c5e6d7943af2be3e14a ]
        Actual hash: [ 0b0d0d077079e1a7186ec2cc640dbd48aebcefa0e1e0caa476128c6396b33d236cb5e8aaada4e1c8868a5d4aaefd90178206dd1569abab63c84af26d3309e2a5 ]

  The file may have been corrupted in transit.  This can be caused by
  proxies.  If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY
  environment variables to
  "https://user:password@your-proxy-ip-address:port/".

````

**Comment by SamuelMarks** [comment]
Yes I agree I edited the issue with commentary. Fixed with a new commit anyway.

Regardless this PR is not mergeable. Maybe the patch needs to be fixed? - Regardless #17052 fails as it stands so one would have difficulty testing (without first fixing that… I'm assuming)

**Review by ytbigvinny27 — APPROVED** [review]
🤌🏾

**Comment by JonLiu1993** [comment]
@SamuelMarks ,Please run the command _vcpkg x-add-version nanogui --overwrite-version_ and submit

**Comment by Tanishq-Banyal** [comment]
When is this going to merge ?

**Comment by Tanishq-Banyal** [comment]
> ```
> CMake Error at CMakeLists.txt:11 (message):
>   The NanoGUI dependency repositories (GLFW, etc.) are missing! You probably
>   did not clone the project with --recursive.  It is possible to recover by
>   calling "git submodule update --init --recursive"
> ```

That happens because nanoGUI depends on other libs (which are downloaded from other GitHub repos when --recursive option is turned on)

One solution can be that you git clone (with --recursive) to local PC and then create a zip archive and upload somewhere.

Or do some cmake magic (Idk anything about cmake)

**Comment by SamuelMarks** [comment]
Well really they should be explicit dependencies of this port, and vcpkg should be instructed to ignore https://github.com/mitsuba-renderer/nanogui/blob/6b82fb1/.gitmodules

(contributions welcome)

**Comment by dg0yt** [comment]
> vcpkg should be instructed to ignore ...

The error message is from the nanoGUI build system. Someone should ask upstream to enable builds without vendored dependencies.

But for the start, just try to create the directory (mkdir), and build NANOGUI_BUILD_GLFW etc. `OFF`?

**Comment by JackBoosY** [comment]
```
CMake Error at CMakeLists.txt:11 (message):
  The NanoGUI dependency repositories (GLFW, etc.) are missing! You probably
  did not clone the project with --recursive.  It is possible to recover by
  calling "git submodule update --init --recursive"
```
We should completely disable the internal project builds and switch to find these dependencies in vcpkg.

**Comment by SamuelMarks** [comment]
Gave it a first attempt

**Comment by JackBoosY** [comment]
```

**Comment by JackBoosY** [comment]
```
CMake Error at C:/a/1/s/scripts/buildsystems/vcpkg.cmake:772 (_find_package):
  Could not find a package configuration file provided by "glad" with any of
  the following names:

    gladConfig.cmake
    glad-config.cmake

  Add the installation prefix of "glad" to CMAKE_PREFIX_PATH or set
  "glad_DIR" to a directory containing one of the above files.  If "glad"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:200 (find_package)
```

**Comment by JackBoosY** [comment]
```
CMake Error at CMakeLists.txt:317 (target_link_libraries):
  Cannot specify link libraries for target "nanogui" which is not built by
  this project.
```

**Comment by PhoebeHui** [comment]
@SamuelMarks, is work still being done for this PR?

**Comment by SamuelMarks** [comment]
Honestly this is just one of my many attempts to get #17194 resolved. I'm sidetracked by other attempts, would welcome anyone take up this helm.

**Comment by JonLiu1993** [comment]
Pinging @SamuelMarks  for response. Is work still being done for this PR?

**Comment by SamuelMarks** [comment]
Ahh @JonLiu1993 I was hoping someone else would take a look to see what I was doing wrong…

**Comment by JackBoosY** [comment]
Can you please merge to master first?

**Comment by JackBoosY** [comment]
```
CMake Error at CMakeLists.txt:317 (target_link_libraries):
  Cannot specify link libraries for target "nanogui" which is not built by
  this project.
```
It looks like some options are not turned on.

File: ports/nanogui/portfile.cmake [hunk 2]
```diff
@@ -2,12 +2,13 @@ vcpkg_fail_port_install(ON_TARGET "uwp")
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
-    REPO wjakob/nanogui
-    REF e9ec8a1a9861cf578d9c6e85a6420080aa715c03 #Commits on Sep 23, 2019
-    SHA512 36c93bf977862ced2df4030211e2b83625e60a11fc9fdb6c1f2996bb234758331d3f41a7fbafd25a5bca0239ed9bac9c93446a4a7fac4c5e6d7943af2be3e14a
+    REPO mitsuba-renderer/nanogui
```
Comment (JackBoosY) [inline ports/nanogui/portfile.cmake:5]: I noticed you changed the repo. Why?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/nanogui/portfile.cmake (hunk 2 above)
Comment (SamuelMarks) [reply ports/nanogui/portfile.cmake:5]: The repo changed, see README of https://github.com/wjakob/nanogui

**Comment by JackBoosY** [comment]
I have no permission to push my commit to this PR.
So please use this file:
[nanogui.zip](https://github.com/microsoft/vcpkg/files/7160257/nanogui.zip)

**Comment by SamuelMarks** [comment]
Hmm, I'll modify manually:
```
vcpkg (nanogui-oct2020) $ patch -fi nanogui/*.patch
patching file nanogui/fix-dependencies.patch
Hunk #1 FAILED at 11.
Hunk #2 FAILED at 74.
Hunk #3 FAILED at 304.
Hunk #4 FAILED at 345.
Hunk #5 FAILED at 361.
5 out of 5 hunks FAILED -- saving rejects to file nanogui/fix-dependencies.patch.rej
```

PS: I'll invite you to my repo so you can edit directly in future?

**Comment by JackBoosY** [comment]
You forgot to commit the patch file.

**Comment by JackBoosY** [comment]
When building nanogui:x64-windows-static:
```
[35/36] cmd.exe /C "cd . && D:\downloads\tools\cmake-3.21.1-windows\cmake-3.21.1-windows-i386\bin\cmake.exe -E vs_link_dll --intdir=CMakeFiles\nanogui.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe  CMakeFiles\nanogui.dir\src\texture_gl.cpp.obj CMakeFiles\nanogui.dir\src\shader_gl.cpp.obj CMakeFiles\nanogui.dir\src\renderpass_gl.cpp.obj CMakeFiles\nanogui.dir\src\opengl.cpp.obj CMakeFiles\nanogui.dir\nanogui_resources.cpp.obj CMakeFiles\nanogui.dir\src\common.cpp.obj CMakeFiles\nanogui.dir\src\widget.cpp.obj CMakeFiles\nanogui.dir\src\theme.cpp.obj CMakeFiles\nanogui.dir\src\layout.cpp.obj CMakeFiles\nanogui.dir\src\screen.cpp.obj CMakeFiles\nanogui.dir\src\label.cpp.obj CMakeFiles\nanogui.dir\src\window.cpp.obj CMakeFiles\nanogui.dir\src\popup.cpp.obj CMakeFiles\nanogui.dir\src\checkbox.cpp.obj CMakeFiles\nanogui.dir\src\button.cpp.obj CMakeFiles\nanogui.dir\src\popupbutton.cpp.obj CMakeFiles\nanogui.dir\src\combobox.cpp.obj CMakeFiles\nanogui.dir\src\progressbar.cpp.obj CMakeFiles\nanogui.dir\src\slider.cpp.obj CMakeFiles\nanogui.dir\src\messagedialog.cpp.obj CMakeFiles\nanogui.dir\src\textbox.cpp.obj CMakeFiles\nanogui.dir\src\textarea.cpp.obj CMakeFiles\nanogui.dir\src\imagepanel.cpp.obj CMakeFiles\nanogui.dir\src\vscrollpanel.cpp.obj CMakeFiles\nanogui.dir\src\colorwheel.cpp.obj CMakeFiles\nanogui.dir\src\colorpicker.cpp.obj CMakeFiles\nanogui.dir\src\graph.cpp.obj CMakeFiles\nanogui.dir\src\tabwidget.cpp.obj CMakeFiles\nanogui.dir\src\canvas.cpp.obj CMakeFiles\nanogui.dir\src\texture.cpp.obj CMakeFiles\nanogui.dir\src\shader.cpp.obj CMakeFiles\nanogui.dir\src\imageview.cpp.obj CMakeFiles\nanogui.dir\src\traits.cpp.obj  /out:nanogui.dll /implib:nanogui.lib /pdb:nanogui.pdb /dll /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL  /INCREMENTAL:NO /LTCG  opengl32.lib  D:\installed\x64-windows-static\debug\lib\glfw3.lib  D:\installed\x64-windows-static\lib\nanovg.lib  D:\installed\x64-windows-static\debug\lib\glad.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
FAILED: nanogui.dll nanogui.lib 
cmd.exe /C "cd . && D:\downloads\tools\cmake-3.21.1-windows\cmake-3.21.1-windows-i386\bin\cmake.exe -E vs_link_dll --intdir=CMakeFiles\nanogui.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe  CMakeFiles\nanogui.dir\src\texture_gl.cpp.obj CMakeFiles\nanogui.dir\src\shader_gl.cpp.obj CMakeFiles\nanogui.dir\src\renderpass_gl.cpp.obj CMakeFiles\nanogui.dir\src\opengl.cpp.obj CMakeFiles\nanogui.dir\nanogui_resources.cpp.obj CMakeFiles\nanogui.dir\src\common.cpp.obj CMakeFiles\nanogui.dir\src\widget.cpp.obj CMakeFiles\nanogui.dir\src\theme.cpp.obj CMakeFiles\nanogui.dir\src\layout.cpp.obj CMakeFiles\nanogui.dir\src\screen.cpp.obj CMakeFiles\nanogui.dir\src\label.cpp.obj CMakeFiles\nanogui.dir\src\window.cpp.obj CMakeFiles\nanogui.dir\src\popup.cpp.obj CMakeFiles\nanogui.dir\src\checkbox.cpp.obj CMakeFiles\nanogui.dir\src\button.cpp.obj CMakeFiles\nanogui.dir\src\popupbutton.cpp.obj CMakeFiles\nanogui.dir\src\combobox.cpp.obj CMakeFiles\nanogui.dir\src\progressbar.cpp.obj CMakeFiles\nanogui.dir\src\slider.cpp.obj CMakeFiles\nanogui.dir\src\messagedialog.cpp.obj CMakeFiles\nanogui.dir\src\textbox.cpp.obj CMakeFiles\nanogui.dir\src\textarea.cpp.obj CMakeFiles\nanogui.dir\src\imagepanel.cpp.obj CMakeFiles\nanogui.dir\src\vscrollpanel.cpp.obj CMakeFiles\nanogui.dir\src\colorwheel.cpp.obj CMakeFiles\nanogui.dir\src\colorpicker.cpp.obj CMakeFiles\nanogui.dir\src\graph.cpp.obj CMakeFiles\nanogui.dir\src\tabwidget.cpp.obj CMakeFiles\nanogui.dir\src\canvas.cpp.obj CMakeFiles\nanogui.dir\src\texture.cpp.obj CMakeFiles\nanogui.dir\src\shader.cpp.obj CMakeFiles\nanogui.dir\src\imageview.cpp.obj CMakeFiles\nanogui.dir\src\traits.cpp.obj  /out:nanogui.dll /implib:nanogui.lib /pdb:nanogui.pdb /dll /version:0.0 /machine:x64 /nologo    /debug /INCREMENTAL  /INCREMENTAL:NO /LTCG  opengl32.lib  D:\installed\x64-windows-static\debug\lib\glfw3.lib  D:\installed\x64-windows-static\lib\nanovg.lib  D:\installed\x64-windows-static\debug\lib\glad.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
LINK Pass 1: command "C:\PROGRA~2\MICROS~2\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe CMakeFiles\nanogui.dir\src\texture_gl.cpp.obj CMakeFiles\nanogui.dir\src\shader_gl.cpp.obj CMakeFiles\nanogui.dir\src\renderpass_gl.cpp.obj CMakeFiles\nanogui.dir\src\opengl.cpp.obj CMakeFiles\nanogui.dir\nanogui_resources.cpp.obj CMakeFiles\nanogui.dir\src\common.cpp.obj CMakeFiles\nanogui.dir\src\widget.cpp.obj CMakeFiles\nanogui.dir\src\theme.cpp.obj CMakeFiles\nanogui.dir\src\layout.cpp.obj CMakeFiles\nanogui.dir\src\screen.cpp.obj CMakeFiles\nanogui.dir\src\label.cpp.obj CMakeFiles\nanogui.dir\src\window.cpp.obj CMakeFiles\nanogui.dir\src\popup.cpp.obj CMakeFiles\nanogui.dir\src\checkbox.cpp.obj CMakeFiles\nanogui.dir\src\button.cpp.obj CMakeFiles\nanogui.dir\src\popupbutton.cpp.obj CMakeFiles\nanogui.dir\src\combobox.cpp.obj CMakeFiles\nanogui.dir\src\progressbar.cpp.obj CMakeFiles\nanogui.dir\src\slider.cpp.obj CMakeFiles\nanogui.dir\src\messagedialog.cpp.obj CMakeFiles\nanogui.dir\src\textbox.cpp.obj CMakeFiles\nanogui.dir\src\textarea.cpp.obj CMakeFiles\nanogui.dir\src\imagepanel.cpp.obj CMakeFiles\nanogui.dir\src\vscrollpanel.cpp.obj CMakeFiles\nanogui.dir\src\colorwheel.cpp.obj CMakeFiles\nanogui.dir\src\colorpicker.cpp.obj CMakeFiles\nanogui.dir\src\graph.cpp.obj CMakeFiles\nanogui.dir\src\tabwidget.cpp.obj CMakeFiles\nanogui.dir\src\canvas.cpp.obj CMakeFiles\nanogui.dir\src\texture.cpp.obj CMakeFiles\nanogui.dir\src\shader.cpp.obj CMakeFiles\nanogui.dir\src\imageview.cpp.obj CMakeFiles\nanogui.dir\src\traits.cpp.obj /out:nanogui.dll /implib:nanogui.lib /pdb:nanogui.pdb /dll /version:0.0 /machine:x64 /nologo /debug /INCREMENTAL /INCREMENTAL:NO /LTCG opengl32.lib D:\installed\x64-windows-static\debug\lib\glfw3.lib D:\installed\x64-windows-static\lib\nanovg.lib D:\installed\x64-windows-static\debug\lib\glad.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\nanogui.dir/intermediate.manifest CMakeFiles\nanogui.dir/manifest.res" failed (exit code 1120) with the following output:
   Creating library nanogui.lib and object nanogui.exp
LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library
common.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwTerminate
common.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwPollEvents
common.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwInit
common.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetTime
common.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwWaitEvents
common.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetErrorCallback
common.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwWindowShouldClose
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSwapBuffers
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetFramebufferSizeCallback
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSwapInterval
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwGetWindowAttrib
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwGetTime
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetDropCallback
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwGetVideoMode
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwGetFramebufferSize
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwPostEmptyEvent
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwGetWindowSize
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetMouseButtonCallback
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetCursor
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwDestroyCursor
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetKeyCallback
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetScrollCallback
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwCreateWindow
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwWindowHint
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetInputMode
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwMakeContextCurrent
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwShowWindow
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwCreateStandardCursor
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwDestroyWindow
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwGetWin32Window
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetWindowTitle
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwGetPrimaryMonitor
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwGetProcAddress
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwHideWindow
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetCursorPosCallback
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetWindowFocusCallback
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetWindowSize
screen.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetCharCallback
textbox.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwGetClipboardString
textbox.cpp.obj : error LNK2001: unresolved external symbol __imp_glfwSetClipboardString
nanogui.dll : fatal error LNK1120: 40 unresolved externals
```

Please note the unresolved symbols have prefix `__imp_`, which means that you didn't use the static macro of glfw in `target_compile_definitions`.

**Comment by JonLiu1993** [comment]
@SamuelMarks ,Could you look at the error log and consider jack's suggestion？

**Comment by JonLiu1993** [comment]
@SamuelMarks ，any progress?

**Comment by JonLiu1993** [comment]
@SamuelMarks , Is this pr working on?

**Comment by JonLiu1993** [comment]
@SamuelMarks ,This is the error log:
````
:\buildtrees\nanogui\src\d0e9ab6d9d-d8efff4234.clean\include\nanogui/opengl.h(42): fatal error C1083: Cannot open include file: 'GLFW/glfw3.h': No such file or directory
[11/36] C:\PROGRA~2\MICROS~3\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x86\cl.exe   /TP -DNANOGUI_BUILD -DNANOGUI_SHARED -DNANOGUI_USE_OPENGL -DNVG_BUILD -DNVG_SHARED -DNVG_STB_IMAGE_IMPLEMENTATION -ID:\buildtrees\nanogui\x86-windows-dbg -ID:\buildtrees\nanogui\src\d0e9ab6d9d-d8efff4234.clean\include /nologo /DWIN32 /D_WINDOWS /W4 /utf-8 /GR /EHsc /MP  /MP /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  /GL   /D _CRT_SECURE_NO_WARNINGS -std:c++17 /showIncludes /FoCMakeFiles\nanogui.dir\src\imagepanel.cpp.obj /FdCMakeFiles\nanogui.dir\ /FS -c D:\buildtrees\nanogui\src\d0e9ab6d9d-d8efff4234.clean\src\imagepanel.cpp
FAILED: CMakeFiles/nanogui.dir/src/imagepanel.cpp.obj 
C:\PROGRA~2\MICROS~3\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x86\cl.exe   /TP -DNANOGUI_BUILD -DNANOGUI_SHARED -DNANOGUI_USE_OPENGL -DNVG_BUILD -DNVG_SHARED -DNVG_STB_IMAGE_IMPLEMENTATION -ID:\buildtrees\nanogui\x86-windows-dbg -ID:\buildtrees\nanogui\src\d0e9ab6d9d-d8efff4234.clean\include /nologo /DWIN32 /D_WINDOWS /W4 /utf-8 /GR /EHsc /MP  /MP /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  /GL   /D _CRT_SECURE_NO_WARNINGS -std:c++17 /showIncludes /FoCMakeFiles\nanogui.dir\src\imagepanel.cpp.obj /FdCMakeFiles\nanogui.dir\ /FS -c D:\buildtrees\nanogui\src\d0e9ab6d9d-d8efff4234.clean\src\imagepanel.cpp
D:\buildtrees\nanogui\src\d0e9ab6d9d-d8efff4234.clean\include\nanogui/opengl.h(42): fatal error C1083: Cannot open include file: 'GLFW/glfw3.h': No such file or directory
[12/36] C:\PROGRA~2\MICROS~3\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x86\cl.exe   /TP -DNANOGUI_BUILD -DNANOGUI_SHARED -DNANOGUI_USE_OPENGL -DNVG_BUILD -DNVG_SHARED -DNVG_STB_IMAGE_IMPLEMENTATION -ID:\buildtrees\nanogui\x86-windows-dbg -ID:\buildtrees\nanogui\src\d0e9ab6d9d-d8efff4234.clean\include /nologo /DWIN32 /D_WINDOWS /W4 /utf-8 /GR /EHsc /MP  /MP /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  /GL   /D _CRT_SECURE_NO_WARNINGS -std:c++17 /showIncludes /FoCMakeFiles\nanogui.dir\src\label.cpp.obj /FdCMakeFiles\nanogui.dir\ /FS -c D:\buildtrees\nanogui\src\d0e9ab6d9d-d8efff4234.clean\src\label.cpp
FAILED: CMakeFiles/nanogui.dir/src/label.cpp.obj 
````

**Comment by SamuelMarks** [comment]
@JonLiu1993 Ok let me see what I can do. Going to generate some new patch diffs for application also

**Comment by JonLiu1993** [comment]
@SamuelMarks, It looks like CI still has some problems：
https://dev.azure.com/vcpkg/public/_build/results?buildId=64640&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by JonLiu1993** [comment]
@SamuelMarks , any updates?

**Comment by SamuelMarks** [comment]
@JonLiu1993 Yeah I need to spend time to get this port working. In the meantime I merged master and updated to latest nanogui…

**Comment by JonLiu1993** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**